lib/store/task.ts

```diff
@@ -17,7 +17,13 @@ interface TaskStore {
   tasks: TaskWithTags[];
   fetchTasks: () => Promise<void>;
   loadingTasks: boolean;
-  addTask: (text: string, tagIds?: string[], date?: string) => void;
+  // Better pattern then a bunch of optional params??
+  addTask: (
+    text: string,
+    tagIds?: string[],
+    date?: string,
+    priority?: number
+  ) => void;
```

Comment:
This is going to get confusing as all heck trying to remember the order of args here. 

We could use an object approach so that we can explicitly handle updates with named arguments.